docs/utilities/feature_flags.md

```diff
@@ -1,57 +1,603 @@
+
+```json hl_lines="4-11"
+
+{
+  "premium_feature": {
+    "default": false,
+    "rules": {
+      "customer tier equals premium": {
+        "when_match": true,
+        "conditions": [
+          {
+            "action": "EQUALS",
+            "key": "tier",
+            "value": "premium"
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+You can have multiple rules with different names. The powertools will return the first result `when_match` of the
+matching rule configuration or `default` value when none of the rules apply.
+
+#### Conditions
+
+The conditions block is a list of `action`, `key` `value`:
+
+```json
+{
+  "action": "EQUALS",
+  "key": "tier",
+  "value": "premium"
+}
+```
+
+The `action` configuration can have 5 different values: `EQUALS`, `STARTSWITH`, `ENDSWITH`, `IN`, `NOT_IN`. The `key`
+and `value` will be compared to the input from the context parameter.
+
+If you have multiple conditions powertools will evaluate the list of conditions as a logical AND, so all conditions
+needs to be matched to return `when_match` value.
+
+=== "features.json"
+
+    ```json  hl_lines="10-11"
+    {
+      "premium_feature": {
+        "default": false,
+        "rules": {
+          "customer tier equals premium": {
+            "when_match": true,
+            "conditions": [
+              {
+                "action": "EQUALS",
+                "key": "tier",
+                "value": "premium"
+              }
+            ]
+          }
+        }
+      }
+    }
+    ```
+
+=== "app.py"
+
+    ```python  hl_lines="2"
+        feature_flags = FeatureFlags(store=app_config)
+        ctx = {"username": "lessa", "tier": "premium", "location": "NL"}
+
+        has_premium_features: bool = feature_flags.evaluate(name="premium_features",
+                                                            context=ctx,
+                                                            default=False
+    ```
+
+## Advanced
+
+### Adjusting in-memory cache
+
+By default, we cache configuration retrieved from the Store for 5 seconds for performance and reliability reasons.
+
+You can override `cache_seconds` parameter when instantiating the store.
+
+```python hl_lines="7"
+from aws_lambda_powertools.utilities.feature_flags import FeatureFlags, AppConfigStore
+
+app_config = AppConfigStore(
+    environment="dev",
+    application="product-catalogue",
+    name="features",
+    cache_seconds=300
+)
+```
+
+### Envelope
+
+There are scenarios where you might want to include feature flags as part of an existing application configuration.
+
+For this to work, you need to use a JMESPath expression via the `envelope` parameter to extract that key as the feature flags configuration.
+
+=== "app.py"
+
+	```python hl_lines="7"
+	from aws_lambda_powertools.utilities.feature_flags import FeatureFlags, AppConfigStore
 
+	app_config = AppConfigStore(
+		environment="dev",
+		application="product-catalogue",
+		name="configuration",
+		envelope = "features"
```

Comment:
shouldn't the envelope value be `feature_flags`? Alternatively we can change the key in`configuration.json` example from `feature_flags` to `features`